### ISSUE TITLE
[BUG] websocket on compress close duplicate

### ISSUE THREAD

**Issue by t0jc** [gorilla/websocket#859]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

in this line 
https://github.com/gorilla/websocket/blob/ac0789be11725ab2285233e9a3800c2312cff4fc/compression.go#L138

and this line
https://github.com/gorilla/websocket/blob/ac0789be11725ab2285233e9a3800c2312cff4fc/conn.go#L1024

### Expected Behavior

if close duplicate will report this log 

` 2023/10/30 21:56:02 websocket: discarding reader close error: io: read/write on closed pipe`

### Steps To Reproduce

1. create websocket link with "EnableCompression"
2. use this websocket write someting
3. in server side got log "websocket: discarding reader close error: io: read/write on closed pipe"

### Anything else?

2023/10/30 21:56:02 websocket: discarding reader close error: io: read/write on closed pipe

**Comment by DVKunion** [comment]
the same +1, any solutions or temporary repair suggestions? :)

**Comment by davidnewhall** [comment]
Disabling compression does not get rid of the error. The error seems triggered by keep alives because it happens without any other requests.

**Comment by jaitaiwan** [comment]
What would really help with moving forward with this bug is having reproduction steps. Then we can just run a go debugger and poke at the stacks.

**Comment by davidnewhall** [comment]
Sounds a bit beyond my skill, but I'd be happy to work with someone to figure this out. 

I have [an app](http://unstable.notifiarr.app) that reproduces this error consistently. I have thousands of users, and I'd like to get this figured out before I release an app update with v1.5.1 of this library (which adds the new log message). If you download the app, you'll also need a (free) account at https://notifiarr.com. There's no (email or anything) verification so you can remain 100% anonymous there.

**Comment by GreenMarmot** [comment]
This issue (along with others) was introduced in commit 666c197fc9157896b57515c3a3326c3f8c8319fe.  The commit message gives no reason for the change.  Fix by reverting the changes around the lines of code linked in the issue.

**Comment by jaitaiwan** [comment]
Thanks for the feedback folks. We recognise that we made a mistake in merging some of these changes and we're working on getting new PRs written and reviewed to fix this going forward.

**Comment by DVKunion** [comment]
> Disabling compression does not get rid of the error. The error seems triggered by keep alives because it happens without any other requests. I sure wish I could pass in a custom logger for this so I can tune the error and figure out which thread it comes from.

Yes, this is just a temporary solution. I have abandoned comrpress to ensure that my logs have no erroneous outputs.If there is a better fix method, please remind me :)

**Comment by elepedus** [comment]
Did anyone manage to resolve this issue? We're using the library to proxy websocket connections, and would love a fix 🙏

**Comment by jaitaiwan** [comment]
We’ve done a revert and plan to do a temporary release to cover things I believe .

**Comment by elepedus** [comment]
Oh awesome! When do you expect to release the fixed version?

**Comment by jaitaiwan** [comment]
We’re talking about internally how to version it, once the decision is made we’ll do it straight away. In the meantime master represents that future version.

**Comment by elepedus** [comment]
Thank you! 🙏

**Comment by conbanwa** [comment]
same issue on @1.5.1, and is was just ok on @1.5.0  version

**Comment by hulkingshtick** [comment]
Was this fixed?  There are no calls to the log package on the main branch.

**Comment by jaitaiwan** [comment]
I believe so - v1.5.3 should be safe now.

**Comment by genticflowlabs** [comment]
Can confirm that v1.5.3 fixed the issue.
